plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2935,17 +2939,25 @@ public MaterializedViewFreshness getMaterializedViewFreshness(ConnectorSession s
         String dependsOnTables = Optional.ofNullable(icebergTable.currentSnapshot())
                 .map(snapshot -> snapshot.summary().getOrDefault(DEPENDS_ON_TABLES, ""))
                 .orElse("");
+        Optional<Long> firstTableChange = Optional.of(Long.MAX_VALUE);
         if (dependsOnTables.isEmpty()) {
+            // If the source of the MV is a PTF then source tables will be empty but source functions won't
+            String dependsOnFunctions = Optional.ofNullable(icebergTable.currentSnapshot())
+                    .map(snapshot -> snapshot.summary().getOrDefault(DEPENDS_ON_FUNCTIONS, ""))
+                    .orElse("");
+            if (!dependsOnFunctions.isEmpty()) {
+                return new MaterializedViewFreshness(UNKNOWN, firstTableChange.map(Instant::ofEpochMilli));
```

Comment:
at this point firstTableChange = Long.MAX_VALUE
so using it will result in always fresh MV when grace period is used

use last refresh time (time of last refresh, ~ snapshot time) as `lMaterializedViewFreshness.astFreshTime` parameter